pkg/cli/admin/ocpcertificates/certgraphanalysis/certkeypair.go

```diff
@@ -0,0 +1,213 @@
+package certgraphanalysis
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"os"
+
+	"github.com/openshift/oc/pkg/cli/admin/ocpcertificates/certgraphapi"
+	"k8s.io/apimachinery/pkg/util/duration"
+)
+
+func toCertKeyPair(certificate *x509.Certificate) (*certgraphapi.CertKeyPair, error) {
+	ret := &certgraphapi.CertKeyPair{
+		Name: fmt.Sprintf("%v::%v", certificate.Subject.CommonName, certificate.SerialNumber),
+		Spec: certgraphapi.CertKeyPairSpec{
+			SecretLocations: nil,
+			OnDiskLocations: nil,
+			CertMetadata:    toCertKeyMetadata(certificate),
+		},
+	}
+
+	details, err := toCertKeyPairDetails(certificate)
+	ret.Spec.Details = details
+	if err != nil {
+		ret.Status.Errors = append(ret.Status.Errors, err.Error())
+	}
+
+	return ret, nil
+}
+
+func toCertKeyPairDetails(certificate *x509.Certificate) (certgraphapi.CertKeyPairDetails, error) {
+	count := 0
+	isClient := false
+	isServing := false
+	isSigner := (certificate.KeyUsage & x509.KeyUsageCertSign) != 0
```

Comment:
wouldn't `certificate.IsCA` work here?

Reply:
>wouldn't certificate.IsCA work here?

I don't know.  Test before a followup to switch.  Also test in cluster-debug-tools please.